packages/context/src/inject.ts

```diff
@@ -249,28 +250,12 @@ function resolveByTag(
 ) {
   const tag: string | RegExp = injection.metadata!.tag;
   const bindings = ctx.findByTag(tag);
-  const values: BoundValue[] = new Array(bindings.length);
 
-  // A closure to set a value by index
-  const valSetter = (i: number) => (val: BoundValue) => (values[i] = val);
-
-  let asyncResolvers: PromiseLike<BoundValue>[] = [];
-  // tslint:disable-next-line:prefer-for-of
-  for (let i = 0; i < bindings.length; i++) {
+  return resolveList(bindings, (binding, result, i) => {
     // We need to clone the session so that resolution of multiple bindings
     // can be tracked in parallel
-    const val = bindings[i].getValue(ctx, ResolutionSession.fork(session));
-    if (isPromise(val)) {
-      asyncResolvers.push(val.then(valSetter(i)));
-    } else {
-      values[i] = val;
-    }
-  }
-  if (asyncResolvers.length) {
-    return Promise.all(asyncResolvers).then(vals => values);
-  } else {
-    return values;
-  }
+    return bindings[i].getValue(ctx, ResolutionSession.fork(session));
```

Comment:
I think `bindings[i]`  can be simplified to `binding`.

The names `bindings` and `binding` are too easy to confuse, please change one of the variables to a more distinctive name.

I would personally simplify the call of `resolveList`:

```ts
return resolveList(
  bindings,
  b => b.getValue(ctx, ResolutionSession.fork(session)),
);
```